tests/admin_views/tests.py

```diff
@@ -2062,6 +2062,13 @@ def test_protected(self):
             '<li>Answer: <a href="%s">Yes.</a></li>' % reverse('admin:admin_views_answer_change', args=(a2.pk,))
         )
 
+    def test_post_delete_protected(self):
+        q = Question.objects.create(question="Why?")
+        Answer.objects.create(question=q, answer="Because.")
+
+        self.client.post(reverse('admin:admin_views_question_delete', args=(q.pk,)), {'post': 'yes'})
```

Comment:
Implementation looks okay but should still check the response here and check that the "protected" page is displayed.